crates/data_components/src/databricks/sql_warehouse.rs

```diff
@@ -345,6 +348,24 @@ struct ExternalLink {
     next_chunk_internal_link: Option<String>,
 }
 
+// Helper function to find the index of the matching closing '>' for a given '<'
+fn find_matching_closing_bracket(input: &str, start: usize) -> Option<usize> {
+    let mut depth = 0;
+    for (i, c) in input[start..].char_indices() {
+        match c {
+            '<' => depth += 1,
+            '>' => {
+                depth -= 1;
+                if depth == 0 {
+                    return Some(start + i);
+                }
+            }
+            _ => {}
+        }
+    }
+    None // No matching '>' found
+}
+
 fn map_databricks_type(type_name: &str) -> Result<DataType, Error> {
```

Comment:
Can we do `SELECT * from table_name LIMIT 0` to get records batch schema?

Reply:
This approach won't work for empty table (no records)

Reply:
Yeah, we should have native schema inference that works with no records.